test/integration/coscheduling_test.go

```diff
@@ -0,0 +1,286 @@
+	for i := min - 2; i >= 0; i-- {
+		podInfo := testCtx.Scheduler.NextPod()
+		pgName2, _ := coscheduling.GetPodGroupLabels(podInfo.Pod)
+		if pgName1 != pgName2 {
+			t.Errorf("PodGroup %q/%q is popped out as expected %v.", pgName2, podInfo.Pod.Name, pgName1)
+		} else {
+			t.Logf("PodGroup %q/%q is popped out as expected.", pgName2, podInfo.Pod.Name)
+		}
+	}
+
+	// Get the second podGroup
+	podInfo = testCtx.Scheduler.NextPod()
+	pgName1, min = coscheduling.GetPodGroupLabels(podInfo.Pod)
+	// Expect Pods are popped in the Coscheduling class order.
+	for i := min - 2; i >= 0; i-- {
+		podInfo := testCtx.Scheduler.NextPod()
+		pgName2, _ := coscheduling.GetPodGroupLabels(podInfo.Pod)
+		if pgName1 != pgName2 {
+			t.Errorf("PodGroup %q/%q is popped out as expected %v.", pgName2, podInfo.Pod.Name, pgName1)
+		} else {
+			t.Logf("PodGroup %q/%q is popped out as expected.", pgName2, podInfo.Pod.Name)
+		}
+	}
+}
+
+func TestCoschedulingPlugin(t *testing.T) {
+	registry := framework.Registry{coscheduling.Name: coscheduling.New}
+	profile := schedapi.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedapi.Plugins{
+			QueueSort: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+			PreFilter: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+			},
+			Permit: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+			},
+			Unreserve: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: coscheduling.Name},
+				},
+			},
+		},
+	}
+	testCtx := util.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestMaster(t, "sched-coscheduling", nil),
+		false,
+		scheduler.WithProfiles(profile),
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+	defer testutils.CleanupTest(t, testCtx)
+
+	cs, ns := testCtx.ClientSet, testCtx.NS.Name
+	// Create a Node.
+	nodeName := "fake-node"
+	node := st.MakeNode().Name("fake-node").Label("node", nodeName).Obj()
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(300, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(300, resource.DecimalSI),
+	}
+	node, err := cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName, err)
+	}
+
+	// Create 6 Pods belongs to two podGroup.
+	// pg1 has three pods, and the resources of the cluster can't meet the min of pg1.
+	// pg2 has three pods, but the resources of the cluster can meet the min of pg2.
+	var pods []*v1.Pod
+	pgNames1 := []string{"pod1-1", "pod1-2", "pod1-3"}
+	pgNames2 := []string{"pod2-1", "pod2-2", "pod2-3"}
+	pause := imageutils.GetPauseImageName()
+	for i := 0; i < len(pgNames1); i++ {
```

Comment:
Current logic creates the Pods in a zigzag order: "pg1-1, pg2-1, pg1-2, pg2-2, ...". However, if I changed the order to "pg1-1, pg1-2, pg1-3, pg2-1, pg2-2, pg2-3", i.e., like this:

```diff
diff --git a/test/integration/coscheduling_test.go b/test/integration/coscheduling_test.go
index a9811e4..a26c992 100644
--- a/test/integration/coscheduling_test.go
+++ b/test/integration/coscheduling_test.go
@@ -221,8 +221,10 @@ func TestCoschedulingPlugin(t *testing.T) {
 			coscheduling.PodGroupMinAvailable: "3",
 		}
 		pods = append(pods, pod)
+	}
 
-		pod = st.MakePod().Namespace(ns).Name(pgNames2[i]).Container(pause).Obj()
+	for i := 0; i < len(pgNames2); i++ {
+		pod := st.MakePod().Namespace(ns).Name(pgNames2[i]).Container(pause).Obj()
 		pod.Spec.Containers[0].Resources = v1.ResourceRequirements{
 			Requests: v1.ResourceList{
 				v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
@@ -239,7 +241,7 @@ func TestCoschedulingPlugin(t *testing.T) {
 	}
 
 	// Create 6 Pods, We will expect them to be scheduled in a reversed order.
-	t.Logf("Start to create 6 Pods.")
+	t.Logf("Start to create %d Pods.", len(pods))
 	for i := range pods {
 		t.Logf("Creating Pod %q", pods[i].Name)
 		_, err := cs.CoreV1().Pods(pods[i].Namespace).Create(context.TODO(), pods[i], metav1.CreateOptions{})
```

The test would fail:

```script
--- FAIL: TestCoschedulingPlugin (17.94s)
    coscheduling_test.go:244: Start to create 6 Pods.
    coscheduling_test.go:246: Creating Pod "pod1-1"
    coscheduling_test.go:246: Creating Pod "pod1-2"
    coscheduling_test.go:246: Creating Pod "pod1-3"
    coscheduling_test.go:246: Creating Pod "pod2-1"
    coscheduling_test.go:246: Creating Pod "pod2-2"
    coscheduling_test.go:246: Creating Pod "pod2-3"
    coscheduling_test.go:265: Failed to schedule Pod "pod2-1": timed out waiting for the condition
    coscheduling_test.go:268: Failed to schedule Pod "pod2-2": timed out waiting for the condition
    coscheduling_test.go:271: Failed to schedule Pod "pod2-3": timed out waiting for the condition
FAIL
FAIL    sigs.k8s.io/scheduler-plugins/test/integration  23.061s
FAIL
```

However, I think the eventual result should be irrelevant, right?